core/src/main/java/org/elasticsearch/common/util/concurrent/AbstractRunnable.java

```diff
@@ -36,7 +36,7 @@ public final void run() {
         try {
             doRun();
         } catch (InterruptedException ex) {
-            Thread.interrupted();
+            Thread.currentThread().interrupt();
```

Comment:
To me that's also the same as above but as we're on the server-side here I'm not as reluctant as on the client side to agree though I still think it's broken. The separate catch block for `InterruptedException` indicates that the original developer was aware that we have to treat `InterruptedException` differently. It's just that the handling is not correct.

However, if we don't handle `InterruptedException` differently, I would remove the separate catch block too:

```
    @Override
    public final void run() {
        try {
            doRun();
        } catch (Throwable t) {
            onFailure(t);
        } finally {
            onAfter();
        }
    }
```